dialect/sql/builder.go

```diff
@@ -2488,6 +2497,9 @@ func (b *Builder) join(qs []Querier, sep string) *Builder {
 		b.WriteString(query)
 		b.args = append(b.args, args...)
 		b.total += len(args)
+		if qe, ok := q.(querierErr); ok {
+			b.AddError(qe.Err())
+		}
```

Comment:
No need to call this method if `Err()` is nil. 
```suggestion
		if qe, ok := q.(querierErr); ok {
			if err := qe.Err(); err != nil {
				b.AddError(err)
			}
		}
```

Reply:
this pr makes AddError ignored nil error, makes code more clear, or do you thing all AddError should be checked ?

Reply:
We can keep the logic in `AddError` that ignores nil errors, but there's no need to call `AddError` on every `join` call if it's unnecessary. Sounds good?

Reply:
Sounds good.